lib/bokken/events.ex

```diff
@@ -488,11 +488,145 @@ defmodule Bokken.Events do
 
   ## Examples
 
-      iex> change_lecture(lecture)
-      %Ecto.Changeset{data: %Lecture{}}
+    iex> change_lecture(lecture)
+    %Ecto.Changeset{}
 
   """
   def change_lecture(%Lecture{} = lecture, attrs \\ %{}) do
     Lecture.changeset(lecture, attrs)
   end
+
+  alias Bokken.Events.Enrollment
+
+  @doc """
+  Returns the list of enrollments.
+
+  ## Examples
+
+    iex> list_enrollments(123)
+      [%Enrollment{}, ...]
+
+    iex> list_enrollments()
+      [%Enrollment{}, ...]
+
+  """
+  def list_enrollments(%{"ninja_id" => ninja_id}) do
+    Enrollment
+    |> where([e], e.ninja_id == ^ninja_id)
+    |> Repo.all()
+    |> Repo.preload([:ninja, :event])
+  end
+
+  def list_enrollments(%{"event_id" => event_id}) do
+    Enrollment
+    |> where([e], e.event_id == ^event_id)
+    |> Repo.all()
+    |> Repo.preload([:ninja, :event])
+  end
+
+  def list_enrollments do
+    Enrollment
+    |> Repo.all()
+    |> Repo.preload([:ninja, :event])
+  end
+
+  @doc """
+  Gets a single enrollment.
+
+  Returns nil if the Enrollment does not exist.
+
+  ## Examples
+
+      iex> get_enrollment(123)
+      %Enrollment{}
+
+      iex> get_enrollment(456)
+      nil
+
+  """
+  def get_enrollment(id, preloads \\ []) do
+    Enrollment
+    |> Repo.get(id)
+    |> Repo.preload(preloads)
+  end
+
+  @doc """
+  Creates an enrollment.
+
+  ## Examples
+
+      iex> create_enrollment(%{field: value})
+      {:ok, %Enrollment{}}
+
+      iex> create_enrollment(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_enrollment(event_id, attrs \\ %{}) do
```

Comment:
What do you think about passing here an `%Event{}` instead of the id, that way you can be sure that the `get_event!` will not be null and pattern matching on the Event.